### PR TITLE
Nest statistics utils functions

### LINKED ISSUE #1293: Move statistics utilities into nested directory structure
As a Developer, I wish the the `statistics_utils.py` had more structure and each function was nested under a utils directory.

### PR DESCRIPTION
**Pull request by dvreed77**
- Nest statistics utils functions
- Closes #1293

### CODE REVIEW

File: docs/source/release_notes.rst [hunk 1]
```diff
@@ -17,6 +17,7 @@ Future Release
     * Changes
         * Remove framework for unused ``woodwork`` CLI (:pr:`1288`)
         * Add back support for Python 3.7 (:pr:`1292`)
+        * Nested statistical utility functions into directory (:pr:`1295`)
```
Comment (thehomebrewnerd) [inline docs/source/release_notes.rst:20]: Need to add your user name to the list of contributors at the bottom of this section.

**Review by thehomebrewnerd — COMMENTED** [review]
One minor release notes thing to fix, but otherwise seems ok to me.

**Comment by tamargrey** [comment]
Are we okay with merging this before #1301 and the 0.13.0 release is out? I the release is waiting on #1299, so maybe it'd be okay to get this in. If we don't, then the release note will just need to be moved to the next future release section.

**Review by thehomebrewnerd — APPROVED** [review]
LGTM!

**Pull request merged**